android/KMEA/app/src/main/java/com/tavultesoft/kmea/cloud/CloudDataJsonUtil.java

```diff
@@ -350,4 +352,38 @@ public static CloudApiTypes.CloudApiReturns retrieveJsonFromDownload(
 
     return null;
   }
+
+  /**
+   * Check if cloudKMP update link is a newer version.
+   * @param updateKMP - Previously stored kmp link. Could be empty if an update wasn't available before
+   * @param cloudKMP - New kmp link from the cloud query
+   * @return boolean if the kmp paths match and cloud kmp version is newer
+   */
+  public static boolean cloudLinkIsNewer(String updateKMP, String cloudKMP) {
+    if (cloudKMP == null || cloudKMP.isEmpty()) {
+      if (!KMManager.isTestMode()) {
+        KMLog.LogError(TAG, "cloudKMP is null or empty");
+      }
+      return false;
+    }
+
+    try {
+      Uri cloudLink = Uri.parse(cloudKMP);
+
+      if (updateKMP == null || updateKMP.isEmpty()) {
+        return true;
+      }
+
+      Uri localLink = Uri.parse(updateKMP);
+
+      boolean pathsMatch = localLink.getLastPathSegment().equalsIgnoreCase(cloudLink.getLastPathSegment());
```

Comment:
If we rename a keyboard using the Cloud deprecation structures, is that handled here? I can't remember...

Reply:
Possibly a source of sadness?  The vai1.1 query returns
```
{
    "keyboards": {
        "vai1.1": {
            "version": "1.1",
            "error": "not available for platform",
            "kmp": "https://keyman.com/go/package/download/keyboard/vai1.1?version=1.1&update=1",
            "deprecatedBy": "athinkra_vai"
        }
    }
}
```

Currently, since the return has "error" field, we'll skip processing the update.

I'm not sure what happens if we were able to update vai1.1

Reply:
Probably not a good choice then -- given it was not an Android-capable one. Which one could we use instead? How about tamil99m?

Reply:
```json
{
    "keyboards": {
        "tamil99m": {
            "version": "1.1",
            "error": "not available as package",
            "deprecatedBy": "basic_kbdtam99"
        }
    }
}
```

and european2 returns
```json
{
    "keyboards": {
        "european2": {
            "version": "1.6",
            "error": "not available as package",
            "deprecatedBy": "sil_euro_latin"
        }
    }
}
```

Reply:
https://api.keyman.com/package-version?platform=android&keyboard=basic_kbdughr
returns
```json
{
    "keyboards": {
        "basic_kbdughr": {
            "version": "1.0",
            "kmp": "https://keyman.com/go/package/download/keyboard/basic_kbdughr?version=1.0&update=1",
            "deprecatedBy": "basic_kbdughr1"
        }
    }
}
```

Reply:
I'll go ahead and merge this. If we need to handle package updates for deprecated keyboards, I'll address in a separate issue.